### DISCUSSION TITLE
Parsing json sparql results

### DISCUSSION THREAD

**Discussion by tayeb83** [eclipse-rdf4j/rdf4j#3367] [Q&A]
Hello,

  I try to connect to an endpoint server which return results in JSON. The problem is when I try to evaluate the results I obtained an SAXParseException

sax.SAXParseException; lineNumber: 2; columnNumber: 1; Content is not allowed in prolog.

How I can avoid this problem ? can I use an sparql json parser directly ?

**Comment by barthanssens** [comment]
Hi,

could you provide a little more info on how you are parsing the results ?
E.g. code snippet where you call the endpoint + start of your processing code. Because it looks like the parser is expecting XML instead of JSON...

Thanks

Bart

**Reply by tayeb83** [reply]
Yes my question is how I can evaluate json results instead of xml? Since the default evaluate (Tuplequery) of rdf4j is expecting XML. 
here is my code : 
` TupleQuery tq = connection.prepareTupleQuery(QueryLanguage.SPARQL,"SELECT * WHERE { ?s ?p ?o }");
			//

			try (TupleQueryResult result = tq.evaluate();) {

				while (result.hasNext()) { 
...`

Is there any way to tell the evaluate that the results are in JSON instead of XML?

**Reply by abrokenjester** [reply]
That is normally something that RDF4J handles for you automatically. RDF4J sends a list of acceptable formats to the server endpoint, and that endpoint should normally indicate in its response what format it is sending back. This is a pretty standard mechanism called 'Content negotiation'. It sounds to me like the endpoint you're querying is sending something unexpected/wrong back in its response. What endpoint are you querying?
